frontend/src/main/components/Students/StudentsTable.js

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import OurTable from "main/components/OurTable";
+
+export default function StudentsTable({ students, currentUser }) {
+
+    const columns = [
+        {
+            Header: 'id',
+            accessor: 'id', // accessor is the "key" in the data
+        },
+        {
+            Header: 'First Name',
+            accessor: 'firstName',
+        },
+        {
+            Header: 'Last Name',
+            accessor: 'lastName',
+        },
+        {
+            Header: 'Perm',
+            accessor: 'perm',
+        }
+    ];
+
+    // Stryker disable ArrayDeclaration : [columns] and [students] are performance optimization; mutation preserves correctness
+    const memoizedColumns = React.useMemo(() => columns, [columns]);
+    const memoizedDates = React.useMemo(() => students, [students]);
```

Comment:
This does not actually memoize these two variables. React only performs a shallow comparison when checking dependency arrays. When React renders this component, it will see that the array reference has changed (as it is a local variable defined in this function) and always return the new (different) array reference.

There are a couple of ways to handle this. We could move the `columns` variable outside of the component (so it becomes more of a constant array, shared across renders), or we could remove `columns` from the dependency array, so React never recomputes `memoizedColumns`.

For `students`, we probably shouldn't handle memoization within this component. This variable has the same problem, where if the `students` array reference changes, it will return the new array (defeating the point of memoization). Instead, the consumer of StudentsTable should make sure the input data is memoized (which will happen automatically if the data is being provided from react-query).

Reply:
Hi Bryan: If you have time to make a branch off of this, and make the changes you suggest, great.   Unfortunately, I don't have time to do that this afternoon and get team03 launched.

Can we make an issue out of those, and go ahead and merge this and move on?

Reply:
I've made an issue here: https://github.com/ucsb-cs156-w22/demo-spring-react-example-v2/issues/94

That may have to be good enough for today.